presto-parquet/src/main/java/com/facebook/presto/parquet/cache/MetadataReader.java

```diff
@@ -87,24 +88,34 @@ public static ParquetFileMetadata readFooter(FSDataInputStream inputStream, Path
         // 4 bytes: MetadataLength
         // MAGIC
 
-        validateParquet(fileSize >= MAGIC.length + PARQUET_METADATA_LENGTH + MAGIC.length, "%s is not a valid Parquet File", file);
-        long metadataLengthIndex = fileSize - PARQUET_METADATA_LENGTH - MAGIC.length;
+        validateParquet(fileSize >= MAGIC.length() + POST_SCRIPT_SIZE, "%s is not a valid Parquet File", file);
 
-        InputStream footerStream = readFully(inputStream, metadataLengthIndex, PARQUET_METADATA_LENGTH + MAGIC.length);
-        int metadataLength = readIntLittleEndian(footerStream);
+        //  EXPECTED_FOOTER_SIZE is an int, so this will never fail
+        byte[] buffer = new byte[toIntExact(min(fileSize, EXPECTED_FOOTER_SIZE))];
+        inputStream.readFully(fileSize - buffer.length, buffer);
+        Slice tailSlice = Slices.wrappedBuffer(buffer);
 
-        byte[] magic = new byte[MAGIC.length];
-        footerStream.read(magic);
-        validateParquet(Arrays.equals(MAGIC, magic), "Not valid Parquet file: %s expected magic number: %s got: %s", file, Arrays.toString(MAGIC), Arrays.toString(magic));
+        Slice magic = tailSlice.slice(tailSlice.length() - MAGIC.length(), MAGIC.length());
+        if (!MAGIC.equals(magic)) {
```

Comment:
either validateParquet(), or
throw ParquetCorruptionException after the if condition

Reply:
I wanted to avoid generating string representations per invocation when the magic values match. Will make it an explicit `throw ParquetCorruptionException` call instead.